lib/smwc_bot/search.ex

```diff
@@ -0,0 +1,68 @@
+defmodule SMWCBot.Search do
+  @moduledoc """
+  Search SMWC for shit.
+  """
+
+  require Logger
+
+  @doc """
+  Search for hack.
+  """
+  @spec for({String.t(), String.t()}) ::
+          {:ok, text :: String.t(), href :: String.t()} | {:ok, nil} | {:error, String.t()}
+  def for({hack, waiting}) do
+    base_uri = base_uri(waiting)
+    filter = URI.encode_query(%{"f[name]" => hack})
+    search_uri = base_uri <> filter
+
+    Logger.debug("Uri = #{search_uri}")
+
+    case HTTPoison.get(search_uri) do
+      {:ok, %{status_code: 200, body: body}} ->
+        body
+        |> Floki.parse_document!()
+        |> Floki.find("div#list_content table tr")
+        |> parse_result_table()
```

Comment:
don't need to handle failure of parse_document! since it will never fail if using the default parser...